cli/pcluster/utils.py

```diff
@@ -766,5 +769,48 @@ def get_cluster_capacity(stack_name):
     )
 
 
+def get_instance_type(instance_type):
+    ec2_client = boto3.client("ec2")
+    try:
+        return ec2_client.describe_instance_types(InstanceTypes=[instance_type]).get("InstanceTypes")[0]
+    except Exception as e:
+        LOGGER.error("Failed when retrieving instance type data for instance type %s", instance_type, e)
+        raise e
+
+
 def is_hit_enabled_cluster(scheduler):
     return scheduler in ["slurm"]
+
+
+def read_remote_file(url):
+    """Read a remote file from an HTTP or S3 url."""
+    try:
+        if urlparse(url).scheme == "s3":
+            match = re.match(r"s3://(.*?)/(.*)", url)
+            bucket, key = match.group(1), match.group(2)
+            file_contents = boto3.resource("s3").Object(bucket, key).get()["Body"].read().decode("utf-8")
+        else:
+            with urllib.request.urlopen("http://www.example.com/") as f:
+                file_contents = f.read().decode("utf-8")
+        return file_contents
+    except Exception as e:
+        LOGGER.error("Failed when reading remote file from url %s", url, e)
```

Comment:
missing %s for error. Can we check all the others since it seems a common mistake (also see above in get_instance_type)

Reply:
Right; sorry for that. I checked all the occurrences in the patch and these two were the last ones.